### PR TITLE
Parse version profile libshaderc

### PR DESCRIPTION
**Pull request by Qining**
Currently the test is not completed, I think we need more test cases.

Didn't add interface in CPP header, I think we don't need one for this method, do we?

### CODE REVIEW

File: libshaderc/src/shaderc_test.cc [hunk 1]
```diff
@@ -956,4 +956,48 @@ TEST_F(CompileKindsTest, TessEvaluation) {
...
+  shaderc_profile expected_profile;
+};
+
+// Test for a helper function to parse version and profile from string.
+using ParseVersionProfileTest =
+    testing::TestWithParam<ParseVersionProfileTestCase>;
+
+TEST_P(ParseVersionProfileTest, FromNullTerminatedString) {
+  const ParseVersionProfileTestCase& test_case = GetParam();
+  int version;
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: good practice to initialize scalar non-class values. (version and profile in this case.)
Some compiler somewhere complain about later uses without initializations.  Just avoid that so we don't convert that warning to an error.

File: libshaderc/src/shaderc_test.cc [hunk 2]
```diff
@@ -956,4 +956,48 @@ TEST_F(CompileKindsTest, TessEvaluation) {
...
+    EXPECT_EQ(test_case.expected_profile, profile);
+  }
+}
+
+INSTANTIATE_TEST_CASE_P(
+    HelperMethods, ParseVersionProfileTest,
+    testing::ValuesIn(std::vector<ParseVersionProfileTestCase>{
+        // Valid version profiles
+        {"450core", true, 450, shaderc_profile_core},
+        {"310es", true, 310, shaderc_profile_es},
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: check a valid result with compatibility profile.

**Comment by dneto0** [comment]
-1  Need another valid test case, and initialize scalars to avoid possible warning of use-before-def.

**Comment by Qining** [comment]
Updated with a compatibility profile test and a empty string test.

**Comment by dneto0** [comment]
+2
I'll squash and merge

File: libshaderc/src/shaderc.cc [hunk 3]
```diff
@@ -443,3 +444,28 @@ void shaderc_get_spv_version(unsigned int* version, unsigned int* revision) {
...
+    case ECompatibilityProfile:
+      *profile = shaderc_profile_compatibility;
+      return true;
+    case ENoProfile:
+      *profile = shaderc_profile_none;
+      return true;
+    case EBadProfile:
+      return false;
+  }
+}
```
Comment (dneto0) [inline libshaderc/src/shaderc.cc]: GCC complains error: control reaches end of non-void function

**Comment by dneto0** [comment]
Squashed and pushed into master as c73f444d05

**Pull request closed without merging**